src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -86,10 +98,19 @@ protected override TestEngineResult LoadPackage()
             foreach (var subPackage in packages)
             {
                 var testFile = subPackage.FullName;
+
+                if (_assemblyResolver != null && !TestDomain.IsDefaultAppDomain()
+                    && subPackage.GetSetting(PackageSettings.ImageRequiresDefaultAppDomainAssemblyResolver, false))
+                {
+                    _assemblyResolver.AddPathFromFile(testFile);
+                }
+
                 IFrameworkDriver driver = driverService.GetDriver(TestDomain, testFile);
                 driver.ID = TestPackage.ID;
                 result.Add(driver.Load(testFile, subPackage.Settings));
                 _drivers.Add(driver);
+
+                _packageIndex.Add(driver, subPackage);
```

Comment:
Actually, does this contradict the idea of sharing a single, central resolver or does it mesh well with it? Once your central resolver is set up for dependencies of extension assemblies, will we be removing all resolution locations after running all the tests in that context as well, or will we need to separately keep track of which paths to remove in each case (or otherwise not share the resolver)?